packages/fuels-core/src/abi_decoder.rs

```diff
@@ -325,6 +332,17 @@ fn skip(slice: &[u8], num_bytes: usize) -> Result<&[u8]> {
     }
 }
 
+fn calculate_num_of_elements(param_type: &ParamType, bytes: &[u8]) -> Result<usize> {
+    let memory_size = param_type.compute_encoding_width() * WORD_SIZE;
+    if bytes.len() % memory_size != 0 {
+        return Err(error!(
+            InvalidData,
+            "The bytes provided do not correspond to a Vec<{:?}> got: {:?}", param_type, bytes
```

Comment:
Update this error message to reflect usage in `decode_raw_slice`.

Reply:
This particular thing is changed in #868 for a more general purpose.